lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -80,6 +80,49 @@ static Type lowerType(Type type) {
   return {};
 }
 
+/// This verifies that the target operation has been lowered to a legal
+// operation.  This checks that the operation recursively has no FIRRTL
+// operations or types.
+static LogicalResult verifyOpLegality(Operation *op) {
+  SmallVector<Operation *> worklist = {op};
+  while (!worklist.empty()) {
+    op = worklist.pop_back_val();
+
+    // Check that this operation is not a FIRRTL op.
+    if (op->getDialect() && (FIRRTLDialect::getDialectNamespace() ==
+                             op->getDialect()->getNamespace()))
+      return op->emitError("Found unhandled FIRRTL operation '")
+             << op->getName() << "'";
+
+    // Helper to check a TypeRange for any FIRRTL types.
+    auto checkTypes = [&](TypeRange types) -> LogicalResult {
+      if (llvm::any_of(types, [](Type type) {
+            return type.getDialect().getNamespace() ==
+                   FIRRTLDialect::getDialectNamespace();
```

Comment:
I'd recommend just using `isa<FIRRTLType>(`

Reply:
So there are some types in the FIRRTL dialect that don't inherit from from `FIRRTLType`, namely the memory port types `cmemory` and `cmemoryport`.  They aren't the sorts of types we're too worried about showing up in other ops, do you think we should protect against this?